docs/api/apply/apply.go

```diff
@@ -0,0 +1,75 @@
+package apply
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/cli-utils/docs/api/event"
+	"sigs.k8s.io/cli-utils/docs/api/inventory"
+	"sigs.k8s.io/cli-utils/docs/api/provider"
+	"sigs.k8s.io/cli-utils/pkg/common"
+)
+
+// NewApplier creates a new applier based on the given provider.
+func NewApplier(provider provider.Provider) *applier {
+	return nil
+}
+
+// applier can apply resources to a cluster. Each instance of the applier
+// can be used with a single cluster and using a single strategy for how
+// objects for the inventory should be created.
+type applier struct {
+
+}
+
+// Options defines the different properties that can be set to adjust
+// the behavior of a call to Apply.
+type Options struct {
+	// ReconcileTimeout defines whether the applier should wait
+	// until all applied resources have been reconciled, and if so,
+	// how long to wait.
+	ReconcileTimeout time.Duration
+
+	// PollInterval defines how often we should poll for the status
+	// of resources.
+	PollInterval time.Duration
+
+	// EmitStatusEvents defines whether status events should be
+	// emitted on the eventChannel to the caller.
+	EmitStatusEvents bool
+
+	// NoPrune defines whether pruning of previously applied
+	// objects should happen after apply.
+	NoPrune bool
```

Comment:
It's tempting to turn this and EmitStatusEvents into enums (like DryRunStrategy) to allow for more prune and status emission styles.

Reply:
Yeah, that is an interesting suggestion. One of the challenges with using boolean variable, is that we can't distinguish between the value being false and the caller not having specified it at all, which makes defaulting fields harder. If the default value of the enum is something like `Unknown..`, it would solve this issue.